New Error ->error[E0599]: no method named `as_mut_ptr` found for struct `AtomicU64` in the current scope

Some of the details are:

info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'

  nightly-x86_64-unknown-linux-gnu unchanged - rustc 1.69.0-nightly (31f858d9a 2023-02-28)

info: checking for self-updates
    Updating git repository `https://github.com/Kindelia/HVM.git`
  Installing hvm v1.0.2 (https://github.com/Kindelia/HVM.git#573e9e2d)
    Updating crates.io index
.
.

Compiling instant v0.1.12
   Compiling hvm v1.0.2 (/home/verbus/.cargo/git/checkouts/hvm-a2339bb8b9261392/573e9e2)
error[E0599]: no method named `as_mut_ptr` found for struct `AtomicU64` in the current scope
   --> src/runtime/base/memory.rs:450:54
    |
450 |         let val = heap.node.get_unchecked(*lvar.next.as_mut_ptr() as usize).load(Ordering::Relaxed);
    |                                                      ^^^^^^^^^^ help: there is a method with a similar name: `as_ptr`



For more information about this error, try `rustc --explain E0599`.
error: could not compile `hvm` due to 14 previous errors
error: failed to compile `hvm v1.0.2 (https://github.com/Kindelia/HVM.git#573e9e2d)`, intermediate artifacts can be found at `/tmp/cargo-installsTN1UZ`


This was the script the above was run from:

#!/usr/bin/env bash
rustup install nightly
cargo +nightly install --force --git https://github.com/Kindelia/HVM.git

--- a/src/runtime/base/memory.rs
+++ b/src/runtime/base/memory.rs
@@ -444,10 +444,10 @@ pub fn alloc(heap: &Heap, tid: usize, arity: u64) -> u64 {
       loop {
         //count += 1;
         //if tid == 9 && count > 5000000 {
-        //println!("[9] slow-alloc {} | {}", count, *lvar.next.as_mut_ptr());
+        //println!("[9] slow-alloc {} | {}", count, *lvar.next.as_ptr());
         //}
         // Loads value on cursor
-        let val = heap.node.get_unchecked(*lvar.next.as_mut_ptr() as usize).load(Ordering::Relaxed);
+        let val = heap.node.get_unchecked(*lvar.next.as_ptr() as usize).load(Ordering::Relaxed);
         // If it is empty, increment length
         if val == 0 {
           length += 1;
@@ -456,11 +456,11 @@ pub fn alloc(heap: &Heap, tid: usize, arity: u64) -> u64 {
           length = 0;
         };
         // Moves cursor right
-        *lvar.next.as_mut_ptr() += 1;
+        *lvar.next.as_ptr() += 1;
         // If it is out of bounds, warp around
-        if *lvar.next.as_mut_ptr() >= *lvar.amax.as_mut_ptr() {
+        if *lvar.next.as_ptr() >= *lvar.amax.as_ptr() {
           length = 0;
-          *lvar.next.as_mut_ptr() = *lvar.amin.as_mut_ptr();
+          *lvar.next.as_ptr() = *lvar.amin.as_ptr();
         }
         // If length equals arity, allocate that space
         if length == arity {
@@ -468,9 +468,9 @@ pub fn alloc(heap: &Heap, tid: usize, arity: u64) -> u64 {
           //println!("[{}] alloc {} at {}", lvar.tid, arity, lvar.next - length);
           //lvar.used.fetch_add(arity as i64, Ordering::Relaxed);
           //if tid == 9 && count > 50000 {
-          //println!("[{}] allocated {}! {}", 9, length, *lvar.next.as_mut_ptr() - length);
+          //println!("[{}] allocated {}! {}", 9, length, *lvar.next.as_ptr() - length);
           //}
-          return *lvar.next.as_mut_ptr() - length;
+          return *lvar.next.as_ptr() - length;
         }
       }
     }

--- a/src/runtime/base/program.rs
+++ b/src/runtime/base/program.rs
@@ -155,10 +155,10 @@ pub fn alloc_body(heap: &Heap, prog: &Program, tid: usize, term: Ptr, vars: &[Ru
           get_var(heap, term, vars.get_unchecked(*index as usize))
         },
         RuleBodyCell::Ptr { value, targ, slot } => {
-          let mut val = value + *aloc.get_unchecked(*targ as usize).as_mut_ptr() + slot;
+          let mut val = value + *aloc.get_unchecked(*targ as usize).as_ptr() + slot;
           // should be changed if the pointer format changes
           if get_tag(*value) <= DP1 {
-            val += (*lvar.dups.as_mut_ptr() & 0xFFF_FFFF) * EXT;
+            val += (*lvar.dups.as_ptr() & 0xFFF_FFFF) * EXT;
           }
           val
         }
@@ -171,25 +171,25 @@ pub fn alloc_body(heap: &Heap, prog: &Program, tid: usize, term: Ptr, vars: &[Ru
     let aloc = &heap.aloc[tid];
     let lvar = &heap.lvar[tid];
     for i in 0 .. nodes.len() {
-      *aloc.get_unchecked(i).as_mut_ptr() = alloc(heap, tid, (*nodes.get_unchecked(i)).len() as u64);
+      *aloc.get_unchecked(i).as_ptr() = alloc(heap, tid, (*nodes.get_unchecked(i)).len() as u64);
     };
-    if *lvar.dups.as_mut_ptr() + dupk >= (1 << 28) {
-      *lvar.dups.as_mut_ptr() = 0;
+    if *lvar.dups.as_ptr() + dupk >= (1 << 28) {
+      *lvar.dups.as_ptr() = 0;
     }
     for i in 0 .. nodes.len() {
-      let host = *aloc.get_unchecked(i).as_mut_ptr() as usize;
+      let host = *aloc.get_unchecked(i).as_ptr() as usize;
       for j in 0 .. (*nodes.get_unchecked(i)).len() {
         let cell = (*nodes.get_unchecked(i)).get_unchecked(j);
         let ptr = cell_to_ptr(heap, lvar, aloc, term, vars, cell);
         if let RuleBodyCell::Var { .. } = cell {
           link(heap, (host + j) as u64, ptr);
         } else {
-          *heap.node.get_unchecked(host + j).as_mut_ptr() = ptr;
+          *heap.node.get_unchecked(host + j).as_ptr() = ptr;
         }
       }
     }
     let done = cell_to_ptr(heap, lvar, aloc, term, vars, cell);
-    *lvar.dups.as_mut_ptr() += dupk;
+    *lvar.dups.as_ptr() += dupk;
     //println!("result: {}\n{}\n", show_ptr(done), show_term(heap, prog, done, 0));
     return done;
   }

